pkg/util/gogoprotomarshal/protomarshal.go

```diff
@@ -92,3 +99,13 @@ func ApplyYAML(yml string, pb proto.Message) error {
 	}
 	return ApplyJSON(string(js), pb)
 }
+
+// ApplyYAML unmarshals a YAML string into a proto message.
+// Unknown fields are notallowed.
+func ApplyYAMLStrict(yml string, pb proto.Message) error {
```

Comment:
Why not add a flag instead of a new function? Since go has no defaults for function args, you might have to change a lot of places but code duplication here is not so awesome.

Reply:
I don't think its better personally. https://martinfowler.com/bliki/FlagArgument.html supports this but I am sure there are others claiming the opposite as well. The code duplication is minimal. We could abstract the implementation out using a non-exported common function but it would be even more code than this almost